db/migrations/20240627171711_allow_null_url.down.sql

```diff
@@ -0,0 +1,3 @@
+BEGIN;
+alter table repositories alter column URL set not null;
+COMMIT;
```

Comment:
I get this error running the down migration

```
{"level":"error","error":"2 errors occurred:\n\t* migration failed: column \"url\" of relation \"repositories\" contains null values in line 0: BEGIN;\nalter table repositories alter column URL set not null;\nCOMMIT;\n (details: pq: column \"url\" of relation \"repositories\" contains null values)\n\t* pq: current transaction is aborted, commands ignored until end of transaction block in line 0: SELECT pg_advisory_unlock($1)\n\n","time":"2024-07-09T13:29:07-04:00","message":"Failed to migrate:"}
{"level":"fatal","error":"database locked","time":"2024-07-09T13:29:07-04:00","message":"Failed to migrate"}
```

Reply:
I had an upload repository created when I ran it

Reply:
so i'm not sure exactly what to do about this.  I guess this may be considered a migration that can't be reverted.   We can't flip that column back to 'not null' whith a null value.  We could delete all repositories (and repo configs, and associated rpms, etc....)  in the migration, but thats not a great solution.  Thoughts?

Reply:
That's true. I can't think of a better way to do it. I guess we can consider it an un-revertable migration.